src/reducers/general.js

```diff
@@ -40,12 +40,16 @@ const general = (state = {
       return Object.assign({}, state, stateUpdate);
     case types.UPDATE_PATHNAME:
       return Object.assign({}, state, {
-        pathname: action.pathname
+        pathname: action.data
       });
     case types.CHANGE_LANGUAGE:
       return Object.assign({}, state, {
         language: action.data
       });
+    case types.CLEAN_START:
```

Comment:
This approach generally makes sense to me. However, I think we might want to set the `state.general.defaults.language` according to the `displayDefaults` as well as setting the `state.general.language`, just since I think it would make sense to keep track of what the default was according to the JSON settings, even if we do have a query setting the language to something else. I would imagine it to look something like this:
```suggestion
    case types.CLEAN_START:
      const defaultLanguage = action.metadata.displayDefaults["language"] || defaults.language;
      return Object.assign({}, state, {
        defaults: Object.assign({}, state.defaults, {language: defaultLanguage}),
        language: query.lang ? query.lang : defaultLanguage
      });
```